### DISCUSSION TITLE
Settings INI/JSON/etc.

### DISCUSSION THREAD

**Discussion by God-damnit-all** [Sophia-Community/SophiApp#12] [Ideas]
Right now, the program doesn't save any settings. When you do get around to it, I ask that you make it save to the binary's directory. If you'd rather have it use AppData or ProgramData by default, maybe you could make it save to the binary's directory instead if it detects the settings file there (even if it's 0 bytes).

**Comment by farag2** [comment]
The feature to export/import settings (like a preset) will be added only in 2.0. 1.0 will be released somewhere in 22H1.

**Reply by God-damnit-all** [reply]
> The feature to export/import settings (like a preset) will be added only in 2.0. 1.0 will be released somewhere in 22H1.

I mainly wanted to get this request out there before AppData/ProgramData was used unilaterally. I like my apps to store their settings files locally (i.e. "portably") whenever possible.

**Reply by farag2** [reply]
The app doesn't (and won't) store any settings at all. That's I was talking about a preset file.

**Reply by God-damnit-all** [reply]
> The app doesn't (and won't) store any settings at all. That's I was talking about a preset file.

What about selecting between Light theme and Dark theme?

**Reply by farag2** [reply]
It detects the current theme so there is no need to store such setting too )

**Reply by farag2** [reply]
if a user set the dark theme for all apps in Windows why he needs to set the light theme for this app?

**Reply by farag2** [reply]
Anyway, we're pretty far away from 2.0 and the commercial version. This is our goal.
Did you check the UWP apps category? The UWP list isn't hardcoded, but being rendered in real-time. All icons are being taken from local recourses too. Everything is done by the documented method. It seems for the first time between all apps in the internet. :)

**Reply by God-damnit-all** [reply]
> if a user set the dark theme for all apps in Windows why he needs to set the light theme for this app?

Some people use unregistered versions of Windows 10/11 (especially in IT when they need to test machine configurations), and the global app theme can't be changed until it's registered - meaning they're stuck on light theme.

**Reply by farag2** [reply]
I know. I will discuss it after the 1.0 release with my colleague.

**Reply by God-damnit-all** [reply]
But maybe instead of having a settings file stored somewhere, there could be `-darktheme`/`-lighttheme` launch parameters instead...
